src/main/java/fi/aalto/cs/apluscourses/presentation/filter/Filter.java

```diff
@@ -0,0 +1,17 @@
+package fi.aalto.cs.apluscourses.presentation.filter;
+
+import java.util.Optional;
+import org.jetbrains.annotations.NotNull;
+
+public interface Filter {
+  /**
+   * Resolves whether or not the filter applies to an item.
+   *
+   * @param item An item to check.
+   * @return True: this filter applies to the item.
+   *         False: this filter could apply to the item, but does not.
+   *         Empty: this filter is not applicable to the item.
```

Comment:
feels a bit unnatural, maybe `false` and emty should be switched?

Reply:
The explanation is very poor but I guess the values are chosen right. What this means: if our filter is say "strings that end with a period":

`"hello."` -> true
`"hi!"` -> false
`0.123` -> empty

Reply:
In terms of filtering, what is the difference between an item "passing" the filter or the filter not being applicable to the item?